### PR TITLE
Update Key Vault, Search samples using SNIPPET def

### PR DESCRIPTION
**Pull request by heaths**
Resolves Key Vault's and Search's portions of #20431

### CODE REVIEW

**Comment by heaths** [comment]
There were only 3 markdown changes in 3 different files that are actually desirable. The indentation fix might've been a tool bug, but the other two were user error e.g. VS trying to constantly reformat the file and put the code on a new line after `/*@@*/`, and I missed it in review.

File: sdk/keyvault/Microsoft.Azure.KeyVault/tests/SampleSnippets.cs [hunk 1]
```diff
@@ -363,10 +377,16 @@ private async Task SecretsMigrationGuide()
...
             {
-                //@@AzureServiceTokenProvider provider = new AzureServiceTokenProvider();
-                /*@@*/ provider = new AzureServiceTokenProvider();
-                //@@KeyVaultClient client = new KeyVaultClient(
-                /*@@*/ client = new KeyVaultClient(
+#if SNIPPET
+                AzureServiceTokenProvider provider = new AzureServiceTokenProvider();
+#else
+                provider = new AzureServiceTokenProvider();
+#endif
```
Comment (pakrym) [inline sdk/keyvault/Microsoft.Azure.KeyVault/tests/SampleSnippets.cs:384]: Worth merging?

File: sdk/search/Azure.Search.Documents/tests/Samples/Sample06_EncryptedIndex.cs [hunk 2]
```diff
@@ -120,25 +126,35 @@ public async Task CreateDoubleEncryptedIndex()
                     Environment.GetEnvironmentVariable("SEARCH_API_KEY"));
 
                 SearchIndexerClient indexerClient = new SearchIndexerClient(endpoint, credential);
-                /*@@*/ indexerClient = resources.GetIndexerClient();
+#if !SNIPPET
+                indexerClient = resources.GetIndexerClient();
+#endif
```
Comment (pakrym) [inline sdk/search/Azure.Search.Documents/tests/Samples/Sample06_EncryptedIndex.cs:131]: I think we can reorder lines and merge ifs

**Review by pakrym — APPROVED** [review]
LGTM but consider reordering code a bit to allow to merge a few #ifs together.

File: sdk/keyvault/Microsoft.Azure.KeyVault/tests/SampleSnippets.cs (hunk 1 above)
Comment (heaths) [reply sdk/keyvault/Microsoft.Azure.KeyVault/tests/SampleSnippets.cs:384]: Yeah. I did that in some of the latter files but forgot to go back and fix a few of these.

File: sdk/search/Azure.Search.Documents/tests/Samples/Sample06_EncryptedIndex.cs (hunk 2 above)
Comment (heaths) [reply sdk/search/Azure.Search.Documents/tests/Samples/Sample06_EncryptedIndex.cs:129]: There's a few places, yeah (and I fixed a few more), but this one, for example, has ordering dependencies e.g. lines that need this below.

**Pull request merged**